chain-signatures/node/src/protocol/signature.rs

```diff
@@ -421,74 +452,84 @@ impl SignatureManager {
-                let presignature = match presignature_manager.take(presignature_id).await {
-                    Ok(presignature) => presignature,
-                    Err(err @ GenerationError::PresignatureIsGenerating(_)) => {
-                        tracing::warn!(me = ?self.me, presignature_id, "presignature is generating, can't join signature generation protocol");
-                        return Err(err);
-                    }
-                    Err(err @ GenerationError::PresignatureIsMissing(_)) => {
-                        tracing::warn!(me = ?self.me, presignature_id, "presignature is missing, can't join signature generation protocol");
-                        return Err(err);
-                    }
-                    Err(err @ GenerationError::PresignatureIsGarbageCollected(_)) => {
-                        tracing::warn!(me = ?self.me, presignature_id, "presignature is garbage collected, can't join signature generation protocol");
-                        return Err(err);
-                    }
-                    Err(err) => return Err(err),
-                };
-                tracing::info!(me = ?self.me, presignature_id, "found presignature: ready to start signature generation");
-                let generator = match Self::generate_internal(
-                    participants,
-                    self.me,
-                    self.public_key,
-                    presignature,
-                    GenerationRequest {
-                        proposer,
-                        request: request.clone(),
-                        epsilon,
-                        entropy,
-                        request_id,
-                        sign_request_timestamp: Instant::now(),
-                    },
-                    cfg,
-                ) {
-                    Ok(generator) => generator,
-                    Err((presignature, err @ InitializationError::BadParameters(_))) => {
-                        tracing::warn!(sign_request = ?sign_request_identifier, presignature.id, ?err, "failed to start signature generation");
-                        return Err(GenerationError::CaitSithInitializationError(err));
-                    }
-                };
-                let generator = entry.insert(generator);
-                crate::metrics::NUM_TOTAL_HISTORICAL_SIGNATURE_GENERATORS
-                    .with_label_values(&[self.my_account_id.as_str()])
-                    .inc();
-                Ok(&mut generator.protocol)
-            }
-            Entry::Occupied(entry) => Ok(&mut entry.into_mut().protocol),
+
+        let entry = match self.generators.entry(sign_id.clone()) {
+            Entry::Vacant(entry) => entry,
+            Entry::Occupied(entry) => return Ok(&mut entry.into_mut().protocol),
+        };
+
+        let Some((request, our_proposer, subset)) = self.sign_queue.take(sign_id) else {
```

Comment:
By taking it out of the queue, doesn't it mean that if any of the later checks fail (`InvalidProposer`, `BadParameters`, maybe more) will lead to an inconsistent state? We will not have processed the signature but we have removed it from our queue. Future calls will return `WaitingForIndexer`, if I understand it right.

Reply:
@ChaoticTempest I know the PR is merged and closed, just checking if this might be a problem

Reply:
good shout, it was indeed an issue. @ppca fixed it in this https://github.com/sig-net/mpc/pull/226